kythe/java/com/google/devtools/kythe/analyzers/java/AutoValuePlugin.java

```diff
@@ -0,0 +1,320 @@
+    Map<String, GeneratedSymbol> setters = new HashMap<>();
+
+    // Search generated AutoValue class for getters/builder+setters.
+    for (JCTree def : classDef.getMembers()) {
+      if (def instanceof JCMethodDecl) {
+        JCMethodDecl method = (JCMethodDecl) def;
+        String name = method.sym.getSimpleName().toString();
+        if (method.getParameters().isEmpty()
+            && !isVoid(method.getReturnType())
+            && !types.overridesObjectMethod(classDef.sym, method.sym)) {
+          Optional<Symbol> abstractSym =
+              findAbstractMethod((ClassSymbol) autoValueClass.sym, method.sym);
+          if (abstractSym.isPresent()) {
+            getters.put(
+                method.sym.getSimpleName().toString(),
+                GeneratedSymbol.builder()
+                    .setAbstractSym(abstractSym.get())
+                    .setGeneratedSym(method.sym)
+                    .build());
+          }
+        }
+      } else if (def instanceof JCClassDecl) {
+        JCClassDecl subClass = (JCClassDecl) def;
+        JCClassDecl builderClass = findAutoValueDecl((ClassType) subClass.type, BUILDER_ANNOTATION);
+        if (builderClass != null) {
+          // Found builder class
+          autoValue.setBuilderSymbol(
+              GeneratedSymbol.builder()
+                  .setAbstractSym(builderClass.sym)
+                  .setGeneratedSym(subClass.sym)
+                  .build());
+          Scope builderScope = builderClass.type.tsym.members();
+
+          // Search builder for setter methods.
+          for (JCTree builderDef : subClass.getMembers()) {
+            if (builderDef instanceof JCMethodDecl) {
+              JCMethodDecl method = (JCMethodDecl) builderDef;
+              Type retType = ((MethodType) method.type).restype;
+              if (method.getParameters().size() == 1 && builderClass.type.equals(retType)) {
+                Optional<Symbol> abstractSym =
+                    findAbstractMethod((ClassSymbol) builderClass.sym, method.sym);
+                if (abstractSym.isPresent()) {
+                  setters.put(
+                      method.sym.getSimpleName().toString(),
+                      GeneratedSymbol.builder()
+                          .setAbstractSym(abstractSym.get())
+                          .setGeneratedSym(method.sym)
+                          .build());
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    if (getters.isEmpty()) {
+      return Optional.empty();
+    }
+
+    // Getters must either be all prefixed or not prefixed; trim prefixes, if found.
+    if (getters.keySet().stream().allMatch(AutoValuePlugin::hasGetPropertyPrefix)) {
+      getters = trimPropertyNames(getters);
+    }
+    // Separately, setters must either be all prefixed or not prefixed; trim prefixes, if found.
+    if (setters.keySet().stream().allMatch(s -> s.startsWith("set"))) {
+      setters = trimPropertyNames(setters);
```

Comment:
Does this and the trimming need to be more sophisticated? For example, what I define an AutoValue with fields "Set<String> settings()" and/or "String issue()" ?

Reply:
It only trims if they all have prefixes.  If an AutoValue has "setA()", "setB()", and "settings()", then the properties will be "a", "b", and "tings".  If the AutoValue has "a", "b", and "settings", then the names will not be trimmed.  This matches the AutoValueProcessor logic.